apps/nccl/src/nccl.cu

```diff
@@ -644,3 +647,44 @@ NCCL_API ncclResult_t ncclGroupEnd() {
   // Do nothing
   return ncclSuccess;
 }
+
+NCCL_API ncclResult_t ncclCommRegister(const ncclComm_t, void*, size_t, void**) {
+  // TODO: Implementation
+  return ncclSuccess;
+}
+
+NCCL_API ncclResult_t ncclCommDeregister(const ncclComm_t, void*) {
+  // TODO: Implementation
+  return ncclSuccess;
+}
+
+ncclResult_t ncclMemAlloc(void** ptr, size_t size) {
+  // Allocate memory using mscclpp::allocSharedPhysicalCuda
+  std::shared_ptr<char> sharedPtr;
+  try {
+    sharedPtr = mscclpp::allocSharedPhysicalCuda<char>(size);
+    if (sharedPtr == nullptr) {
+      throw mscclpp::Error("ncclMemAlloc failed", mscclpp::ErrorCode::InvalidUsage);
+    }
+  } catch (const mscclpp::Error& e) {
+    std::ostringstream oss;
+    oss << "Mscclpp failure: " << e.what() << std::endl;
```

Comment:
All console messages should be either a log or an exception. This message is neither of them. And it looks unnecessary.

Reply:
Removed this console message